src/trio/_tests/module_with_deprecations.py

```diff
@@ -1,24 +1,27 @@
 regular = "hi"
 
-from .. import _deprecate
-
-_deprecate.enable_attribute_deprecations(__name__)
-
 # Make sure that we don't trigger infinite recursion when accessing module
 # attributes in between calling enable_attribute_deprecations and defining
 # __deprecated_attributes__:
 import sys
+from typing import TYPE_CHECKING
```

Comment:
This comment and its checks aren't relevant anymore. Might be better to pass `globals()` into the function (or just have it lookup in `sys.modules`), then `__deprecated_attributes__` would work the same way. Although currently we don't have any, so it might be fine to require them to be passed directly to the function.

Reply:
Yeah but actually I think maybe setting `__getattr__` on `sys.modules` might be better to avoid having to get `TYPE_CHECKING`.

Reply:
Alright I changed some stuff. Kept the same arguments (I think passing `globals` is a bit much and it's not like we have compatibility guarantees for it) but made it nicer to use / a better name / updated that file.